src/table/utils/selections-utils.js

```diff
@@ -98,26 +103,25 @@ export const getSelectedRows = ({ selectedRows, cell, evt }) => {
 
 const selectCell = (state, payload) => {
   const { api, rows, colIdx } = state;
-  const { cell, announce, evt } = payload;
+  const { cell, announce, evt, isSelectMultiValues } = payload;
   let selectedRows = {};
 
-  if (colIdx === -1) {
-    api.begin('/qHyperCubeDef');
-  } else if (colIdx === cell.colIdx) {
-    selectedRows = { ...rows };
-  } else {
-    return state;
-  }
+  if (colIdx === -1) api.begin('/qHyperCubeDef');
+  else if (colIdx === cell.colIdx) selectedRows = { ...rows };
+  else return state;
 
   selectedRows = getSelectedRows({ selectedRows, cell, evt });
   const selectedRowsLength = Object.keys(selectedRows).length;
   handleAnnounceSelectionStatus({
     announce,
     rowsLength: selectedRowsLength,
-    isAddition: selectedRowsLength > Object.keys(rows).length,
+    isAddition: selectedRowsLength >= Object.keys(rows).length,
   });
 
   if (selectedRowsLength) {
+    if (isSelectMultiValues) {
+      return { ...state, rows: selectedRows, colIdx: cell.colIdx, isSelectMultiValues };
```

Comment:
no need to have two identical return statements here, put the api.select call inside an else instead